### PR TITLE
Java interview Медведев Илья

### PR DESCRIPTION
**Pull request by hedinant**

### CODE REVIEW

File: Task.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+public abstract class Task {
+    private String name;
+    private TaskStatus status;
+    private long schedule;
+    private Set<Task> dependsOn;
+
+    public abstract void runTask(EntityManager entityManager);
```
Comment (hedinant) [inline Task.java:7]: test

File: Task.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        TaskManager.getInstance().skeduleTask(this);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Task)) return false;
+
+        Task task = (Task) o;
+
```
Comment (itgabrut) [inline Task.java:26]: name has a private access in Task; Possible need to use getter instaed

File: TaskHistoryRecord.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+@Entity
+@Table(name = "taskhistory")
+public class TaskHistoryRecord {
+    @Id
+    private long id;
+    private String record;
+
```
Comment (itgabrut) [inline TaskHistoryRecord.java:7]: Entity demands a default constructor with no parameters

File: TaskManager.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    private LinkedList<Task> queue = new LinkedList<>();
+    private HashSet<Task> running = new HashSet<>();
+    private HashSet<Task> blocked = new HashSet<>();
+    private Thread checker = new Thread(() -> {
+        while (true) {
+            checkRun();
+        }
+    });
+
+    public TaskManager() {
```
Comment (itgabrut) [inline TaskManager.java:15]: instance is a static and will be reassigned here each time. Can't get meaning of all it

Comment (hedinant) [reply TaskManager.java:15]: see Task finish, reschedule

File: TaskManager.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                " finished " + System.currentTimeMillis()));
+
+        blocked.stream().filter(task1 -> task1.getDependsOn().contains(task)).forEach(task1 -> {
+            if (Collections.disjoint(task1.getDependsOn(), running)) {
+                blocked.remove(task1);
+                runTask(task1);
+            }
+        });
+    }
+
```
Comment (itgabrut) [inline TaskManager.java:53]: typo in word (schedule)

File: Task.java (hunk 2 above)
Comment (hedinant) [reply Task.java:26]: Idea generating like this, I think it knows the right way.

File: TaskManager.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    public static TaskManager getInstance() {
+        return instance;
+    }
+
+    @Transactional
+    public void runTask(Task task) {
+        if (Collections.disjoint(task.getDependsOn(), running)) {
+            running.add(task);
+            task.setStatus(TaskStatus.Working);
+            new Thread(() -> task.runTask(entityManager)).run();
```
Comment (itgabrut) [inline TaskManager.java:29]: runTask(entityManager) - is not clear

**Pull request closed without merging**